src/slidedown/Slidedown.ts

```diff
@@ -127,83 +135,92 @@ export default class Slidedown {
   /**
    * only used with Category Slidedown
    */
-  setSaveState(state: boolean) {
+  setSaveState(state: boolean): void {
     if (!this.categoryOptions) {
       Log.debug("Slidedown private category options are not defined");
       return;
     }
 
     if (state) {
       // note: savingButtonText is hardcoded in constructor. TODO: pull from config & set defaults for future release
-      this.allowButton.innerHTML = this.getIndicatorHolderHtmlWithText(this.categoryOptions!.savingButtonText!);
-      addDomElement(this.buttonIndicatorHolder, 'beforeend',
-        getLoadingIndicatorWithColor(COLORS.whiteLoadingIndicator));
+      this.allowButton.innerHTML = this.getIndicatorHolderHtmlWithText(this.categoryOptions.savingButtonText).innerHTML;
```

Comment:
nit, use [`replaceChild`](https://developer.mozilla.org/en-US/docs/Web/API/Node/replaceChild) over `innerHTML`.

Reply:
Looking at this again [`replaceWith`](https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/replaceWith) would probably be a better fit in this case.

Reply:
Ah, yep that makes more sense

Reply:
I ended up switching to use `insertAdjacentElement` ([mdn docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentElement)) to individually insert the children nodes. I had to also separate the helper function into two to return the span element and indicator holder separately.